tests/forms_tests/widget_tests/test_radioselect.py

```diff
@@ -28,25 +28,20 @@ def test_nested_choices(self):
             ('Video', (('vhs', 'VHS'), ('dvd', 'DVD'))),
         )
         html = """
-        <ul id="media">
-        <li>
-        <label for="media_0"><input id="media_0" name="nestchoice" type="radio" value="unknown"> Unknown</label>
-        </li>
-        <li>Audio<ul id="media_1">
-        <li>
-        <label for="media_1_0"><input id="media_1_0" name="nestchoice" type="radio" value="vinyl"> Vinyl</label>
-        </li>
-        <li><label for="media_1_1"><input id="media_1_1" name="nestchoice" type="radio" value="cd"> CD</label></li>
-        </ul></li>
-        <li>Video<ul id="media_2">
-        <li><label for="media_2_0"><input id="media_2_0" name="nestchoice" type="radio" value="vhs"> VHS</label></li>
-        <li>
-        <label for="media_2_1">
-        <input checked id="media_2_1" name="nestchoice" type="radio" value="dvd"> DVD
-        </label>
-        </li>
-        </ul></li>
-        </ul>
+        <div id="media">
+        <div>
+        <label for="media_0"><input type="radio" name="nestchoice" value="unknown" id="media_0"> Unknown</label></div>
+        <label>Audio</label>
```

Comment:
I'm less sure about this. I've used a `<label>` here but maybe it should just be `<strong>`? I'm also less sure of the accessibility issues associated with this layout.

Reply:
Hmmm good question. I don’t recall ever seeing nested/grouped radio buttons like this before, would need to research it. My first impression is that the whole field should be a `fieldset` and `legend`, and then I don’t know [how well that nests](https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/) so it might be warranted to use more advanced ARIA so the label of each item within each group is prefixed by the group label.

Something like:

```jinja
<div id="media">
  <label id="media_1_label">Audio</label>
  <div>
    <label for="media_1_0">
      <input aria-labelledby="media_1_label media_1_0_label" type="radio" name="nestchoice" value="vinyl" id="media_1_0" />
      <span id="media_1_0_label">Vinyl</span>
    </label>
  </div>
</div>
```


For the purpose of this PR I think `<label>` is fine. It has no semantic value without being associated to a field that I know of, but neither does `<strong>`, and `label` feels more likely to be something people can add styles to specifically for this pattern without it having unintended repercussions.

Reply:
This is the tricky one. The widget code will need a bit of work to add in the additional `aria-labelledby` attributes and so on. 

Let me see if I can work out how this could work, we can then have a discussion if we want to do that (or not).

Reply:
@felixxm -- could I get your thoughts? 

This line `<span id="media_1_0_label">Vinyl</span>` here is quite a large change. We'd need to change this [template](https://github.com/django/django/blob/main/django/forms/templates/django/forms/widgets/input_option.html) to wrap the `{{ widget.label }}` in some logic, and back it up by passing in the right data to the `context` and so on. This is all fine, but makes this patch _much_ bigger and is for a much smaller use case. 

We'd also probably want to generalise the change so folk could start passing in css classes to the span, and maybe have it as a `div` rather than a `span` for example. Pushing this a bit further you could have it as a `label` and end up with something that is getting much closer to what some css frameworks need (say [bootstrap](https://getbootstrap.com/docs/5.1/forms/checks-radios/)). I could see this as a useful feature, but seems too much to include as part of this PR?

I was therefore wondering, if we could take this now for the 80% (99%?) use case and revisit this later.

(I think I feel a little flustered, as part of me would like to look at a couple of other items ahead of the feature freeze. Also if we go with the larger change I intimated above that would probably take several months effort at the pace I work at to think about alone)